src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

The `HttpExceptionFilter` class's `catch` method seems to be handling exceptions and formatting responses based on the caught exception. It would be beneficial to add error handling for potential exceptions that might occur during the response generation process to ensure robustness.

```diff
+ try {
+     // Existing code block from line 15 to 33
+ } catch (error) {
+     this.logger.error(`An error occurred while processing the exception: ${error}`);
+     response.status(500).json(APIResponse.create(0, 500, 'Internal Server Error'));
+ }
```

This addition introduces a try-catch block to handle any errors that might occur during the response generation process and provides a generic error message in case of such failures.



<!-- This is an auto-generated comment by PullProofAI -->